src/app/clusters/camera-av-settings-user-level-management-server/camera-av-settings-user-level-management-server.cpp

```diff
@@ -0,0 +1,1079 @@
+    Optional<int16_t> tiltDelta = commandData.tiltDelta;
+    Optional<int8_t> zoomDelta  = commandData.zoomDelta;
+    int16_t newPan              = mMptzPosition.pan.Value();
+    int16_t newTilt             = mMptzPosition.tilt.Value();
+    int8_t newZoom              = static_cast<int8_t>(mMptzPosition.zoom.Value());
+
+    // Validate the received command fields
+    //
+    if (panDelta.HasValue())
+    {
+        if (!HasFeature(Feature::kMechanicalPan))
+        {
+            ChipLogError(Zcl, "Mechanical Pan not supported although panDelta value provided");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+            return;
+        }
+        int16_t panDeltaValue = panDelta.Value();
+        if (panDeltaValue > (mPanMax - mPanMin) || panDeltaValue < -(mPanMax - mPanMin))
+        {
+            ChipLogError(Zcl, "PanDelta value received is out of range.");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+            return;
+        }
+
+        newPan = static_cast<int16_t>(newPan + panDeltaValue);
+        if (newPan > mPanMax)
+        {
+            newPan = mPanMax;
+        }
+        if (newPan < mPanMin)
+        {
+            newPan = mPanMin;
+        }
+
+        hasAtLeastOneValue = true;
+    }
+
+    if (tiltDelta.HasValue())
+    {
+        if (!HasFeature(Feature::kMechanicalTilt))
+        {
+            ChipLogError(Zcl, "Mechanical Tilt not supported although tiltDelta value provided");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+            return;
+        }
+        int16_t tiltDeltaValue = tiltDelta.Value();
+        if (tiltDeltaValue > (mTiltMax - mTiltMin) || tiltDeltaValue < -(mTiltMax - mTiltMin))
+        {
+            ChipLogError(Zcl, "TiltDelta value received is out of range.");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+            return;
+        }
+
+        newTilt = static_cast<int16_t>(newTilt + tiltDeltaValue);
+        if (newTilt > mTiltMax)
+        {
+            newTilt = mTiltMax;
+        }
+        if (newTilt < mTiltMin)
+        {
+            newTilt = mTiltMin;
+        }
+
+        hasAtLeastOneValue = true;
+    }
+
+    if (zoomDelta.HasValue())
+    {
+        if (!HasFeature(Feature::kMechanicalZoom))
+        {
+            ChipLogError(Zcl, "Mechanical Zoom not supported although zoomDelta value provided");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+            return;
+        }
+        int8_t zoomDeltaValue = zoomDelta.Value();
+        if (zoomDeltaValue > (mZoomMax - 1) || zoomDeltaValue < -(mZoomMax - 1))
+        {
+            ChipLogError(Zcl, "ZoomDelta value received is out of range.");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+            return;
+        }
+
+        newZoom = static_cast<int8_t>(newZoom + zoomDeltaValue);
```

Comment:
OK, so here our spec constraints are:

* 2 <= ZoomMax <= 100.
* So -99 <= zoomDeltaValue <= 99.
* newZoom <= 100 before the addition.

That means this addition can in fact overflow (because 199 doesn't fit in `int8_t`), leading to incorrect behavior.  We clearly need some tests here.

We need to either make the result of the addition a type large enough to hold any necessary values, or rewrite this logic in some way.  For example:

1. Make newZoom a `uint8_t`.
2. If zoomDeltaValue < 0, check for `1 + static_cast<uint8_t>(-zoomDeltaValue) < newZoom` (before modifying the value of `newZoom`, else just set `newZoom` to `1`.
3. etc.